source/simulator/assembly.cc

```diff
@@ -804,6 +913,14 @@ namespace aspect
                                                parameters.composition_degree)
                                               +
                                               (parameters.stokes_velocity_degree+1)/2),
+                                  //TODO: give this a 1-point QGauss to save space, as this is never going to be used here? (discontinuous temperature field doesn't require stabilization)
+                                  QGauss<dim-1>((advection_field.is_temperature()
```

Comment:
If the advection_field is discontinuous, we're never going to want to add stabilization (not that I've ever come across at least - perhaps some do!). Thus when we call compute_viscosity below, we are always going to be in the context of a continuous field. These don't need a face _quadrature in the scratch object, so it's probably better to minimise the memory and computation in the scratch constructor by passing a minimal quadrature rule. Is that the best way to proceed?